### PR TITLE
Item has two handles after import

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |
## Problem description
If the Item has pre-registered PID it should be unbound before installation.

### CODE REVIEW

**Review by MajoBerger — CHANGES_REQUESTED** [review]
do not condition replacement of handles with this property - up for discission.

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java [hunk 1]
```diff
@@ -317,11 +318,19 @@ public ItemRest importItem(HttpServletRequest request) throws SQLException, Auth
         context.setCurrentUser(eperson);
         context.turnOffAuthorisationSystem();
         WorkspaceItem workspaceItem = workspaceItemService.create(context, collection, false);
+        // created item
+        Item item = workspaceItem.getItem();
+        // if the created item has pre-registered PID and the importing Item has handle which must be imported, the
+        // pre-registered PID must be unbound from the created Item, otherwise the Item will have two handles.
+        if (DSpaceServicesFactory.getInstance().getConfigurationService()
+                .getBooleanProperty("identifiers.submission.register", false) &&
```
[the remark this answers is not in the document]
Comment (milanmajchrak) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java:326]: I do not agree. What if Item already has a handle, we sent desired handle and the property `identifiers.submission.register` is set to false (should not have pre-registered handle) and somehow we need to keep that already assigned handle? This fix addresses only the error we are encountering.

[the remark this answers is not in the document]
Comment (MajoBerger) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java:326]: Is there a scenario, when we need two handles? And if yes, will the second handle be assigned like this? Not afterwards?

**Review by vidiecan — APPROVED** [review]

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (MajoBerger) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java:326]: @vidiecan @milanmajchrak I do not know what is the current situation, I need to know more to resolve. I expect more comments to answer my question.

Marking this for standup tomorrow.

[the remark this answers is not in the document]
Comment (MajoBerger) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java:326]: we agreed verbally, that this solution is ok

**Review by MajoBerger — APPROVED** [review]

**Pull request merged**